src/org/benf/cfr/reader/bytecode/analysis/types/MethodPrototypeAnnotationsHelper.java

```diff
@@ -96,13 +104,22 @@ void dumpParamType(JavaTypeInstance arg, final int paramIdx, Dumper d) {
         dumpAnnotationTableEntries(declAnnotationsToDump, d);
 
         if (typeAnnotationsToDump.isEmpty()) {
-            d.dump(arg);
+            if (isVarargs) {
+                ((JavaArrayTypeInstance) arg).toVarargString(d);
+            } else {
+                d.dump(arg);
+            }
         } else {
             JavaAnnotatedTypeInstance jat = arg.getAnnotatedInstance();
             DecompilerComments comments = new DecompilerComments();
             TypeAnnotationHelper.apply(jat, typeAnnotationsToDump, comments);
             d.dump(comments);
-            d.dump(jat);
+
+            if (isVarargs) {
+                ((JavaArrayTypeInstance.Annotated) jat).dumpVarargs(d);
```

Comment:
I know this is the only type it could be, but twitchy about having this cast here